lcdblib/plotting/results_table.py

```diff
@@ -0,0 +1,1265 @@
+
+
+class DESeqResults(DifferentialExpressionResults):
+    __doc__ = _base_doc % dedent(
+        """
+        Class for working with results from DESeq.
+
+        Just like a DifferentialExpressionResults object, but sets the
+        pval_column, lfc_column, and mean_column to the names used in DESeq
+        (v1) output.
+        """)
+
+    def colormapped_bedfile(self, genome, cmap=None):
+        """
+        Create a BED file with padj encoded as color
+
+        Features will be colored according to adjusted pval (phred
+        transformed).  Downregulated features have the sign flipped.
+
+        Parameters
+        ----------
+        cmap : matplotlib colormap
+            Default is matplotlib.cm.RdBu_r
+
+        Notes
+        -----
+        Requires a FeatureDB to be attached.
+        """
+        if self.db is None:
+            raise ValueError("FeatureDB required")
+        db = gffutils.FeatureDB(self.db)
+
+        def scored_feature_generator(d):
+            for i in range(len(d)):
+                try:
+                    feature = db[d.ix[i]]
+                except gffutils.FeatureNotFoundError:
+                    raise gffutils.FeatureNotFoundError(d.ix[i])
+                score = -10 * np.log10(d.padj[i])
+                lfc = d.log2FoldChange[i]
+                if np.isnan(lfc):
+                    score = 0
+                if lfc < 0:
+                    score *= -1
+                feature.score = str(score)
+                feature = extend_fields(
+                    gff2bed(gffutils.helpers.asinterval(feature)), 9)
+                fields = feature.fields[:]
+                fields[6] = fields[1]
+                fields[7] = fields[2]
+                fields.append(str(d.padj[i]))
+                fields.append(str(d.pval[i]))
+                fields.append('%.3f' % d.log2FoldChange[i])
+                fields.append('%.3f' % d.baseMeanB[i])
+                fields.append('%.3f' % d.baseMeanB[i])
+                yield pybedtools.create_interval_from_list(fields)
+
+        x = pybedtools.BedTool(scored_feature_generator(self)).saveas()
+        norm = x.colormap_normalize()
+        if cmap is None:
+            cmap = cm.RdBu_r
+        cmap = colormap_adjust.cmap_center_point_adjust(
+            cmap, [norm.vmin, norm.vmax], 0)
+
+        def score_zeroer(f):
+            f.score = '0'
+            return f
+        return x.each(add_color, cmap=cmap, norm=norm)\
+                .sort()\
+                .each(score_zeroer)\
+                .truncate_to_chrom(genome)\
+                .saveas()
+
+    def autosql_file(self):
+        """
+        Generate the autosql for DESeq results (to create bigBed)
+
+        Returns a temp filename containing the autosql defining the extra
+        fields.
+
+        This for creating bigBed files from BED files created by
+        colormapped_bed.  When a user clicks on a feature, the DESeq results
+        will be reported.
+        """
```

Comment:
Thanks! Still has issues with corner cases (all up, or all down, or nothing changed) and should expose a Normalize instance config to be able to make colormaps the same across multiple experiments. But it's a nice way of zooming to interesting places in the browser without needing collaborators to dig through the tsv of results.